packages/core/src/codewhispererChat/controllers/chat/telemetryHelper.ts

```diff
@@ -148,13 +148,14 @@ export class CWCTelemetryHelper {
     }
 
     public getContextType(prompt: AdditionalContextPrompt): string {
-        if (prompt.relativePath.startsWith(path.join('.amazonq', 'rules'))) {
-            return 'rule'
-        } else if (prompt.filePath.startsWith(getUserPromptsDirectory())) {
-            return 'prompt'
-        } else {
-            return 'file'
+        if (prompt.filePath.endsWith(promptFileExtension)) {
+            if (prompt.relativePath.startsWith(path.join('.amazonq', 'rules'))) {
+                return 'rule'
+            } else if (prompt.filePath.startsWith(getUserPromptsDirectory())) {
```

Comment:
This condition seems fragile. Was `prompt.filePath` and `getUserPromptsDirectory()`  normalized? Better to use `isInDirectory` or similar. https://github.com/aws/aws-toolkit-vscode/blob/33b526a0264c90072a393efc1a574fd0fae64a02/packages/core/src/shared/filesystemUtilities.ts#L127